timely_beliefs/beliefs/classes.py

```diff
@@ -788,26 +857,102 @@ def rolling_viewpoint_accuracy(
+            if reference_belief_horizon is None:
+                df_observation = belief_utils.select_most_recent_belief(df)
+            else:
+                df_observation = belief_utils.select_most_recent_belief(
+                    df[
+                        df.index.get_level_values("belief_horizon")
+                        >= reference_belief_horizon
+                    ]
+                )
+
+            # If applicable, decide which source provides the observations that are considered to be true when we compute scores
+            if reference_source is not None:
+                df_observation = df_observation.set_event_value_from_source(
+                    reference_source
+                )
+
+        # Combine the forecasts and observations into one DataFrame
+        df_observation = df_observation.drop_belief_time_or_horizon_index_level()
+        df_forecast = df_forecast.drop_belief_time_or_horizon_index_level()
+        df = pd.concat([df_forecast, df_observation], axis=1)
+        df.columns = ["event_value", "reference_value"]
+
+        return belief_utils.compute_accuracy_scores(df, lite_metrics=lite_metrics)
+
+    def plot(
+        self,
+        show_accuracy: bool = False,
+        active_fixed_viewpoint_selector: bool = True,
+        reference_source: BeliefSource = None,
+        intuitive_forecast_horizon: bool = True,
+        interpolate: bool = True,
+    ) -> alt.LayerChart:
+        """Visualize the BeliefsDataFrame in an interactive Altair chart.
+
+        :param show_accuracy: Set to False to plot time series data only
+        :param active_fixed_viewpoint_selector: If true, fixed viewpoint beliefs can be selected
+        :param reference_source: BeliefSource to indicate that the accuracy should be determined with respect to the beliefs held by the given source
+        :param intuitive_forecast_horizon: If true, horizons are shown with respect to event start rather than knowledge time
+        :param interpolate: If True, the time series chart shows a user-friendly interpolated line rather than more accurate stripes indicating average values
+        :returns: Altair chart object with a vega-lite representation (for more information, see reference below).
+
+        >>> chart = df.plot(df.lineage.sources[0])
+        >>> chart.save("chart.json")
+        >>> chart.serve()
+
+        References
+        ----------
+        Altair: Declarative Visualization in Python.
+            https://altair-viz.github.io
+        """
+        return visualization_utils.plot(
+            self,
+            show_accuracy=show_accuracy,
+            active_fixed_viewpoint_selector=active_fixed_viewpoint_selector,
+            reference_source=reference_source,
+            intuitive_forecast_horizon=intuitive_forecast_horizon,
+            interpolate=interpolate,
+        )
+
+    def set_reference_values(
+        self, reference_source: BeliefSource = None
+    ) -> "BeliefsDataFrame":
```

Comment:
This function deserves a docstring (what does it do?) Also, it seems it might be used in `fixed_viewpoint_accuracy` or `rolling_viewpoint_accuracy`? (I might be wrong, but it looks similar to some code in there)

Reply:
I was hoping to refactor this code cleanly, but it proved to be harder than it looked. I'll add comments on what the future should hold for this function, but the actual refactoring will be a separate ticket.

Reply:
Missing support for cleanly concatenating BeliefsDataFrames without loss of metadata was what was holding this back. [This is now fixed.](https://github.com/SeitaBV/timely-beliefs/commit/5f06f89848e42f400b0462ae22c4491c57a8a55b) I then went ahead and did [the actual refactoring[(https://github.com/SeitaBV/timely-beliefs/commit/c266c645970437028596254a62e29deb48881900)